lib/ansible/modules/cloud/vmware/vmware_object_rename.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright: (c) 2018, Abhijeet Kasurde <akasurde@redhat.com>
+# Copyright: (c) 2018, Ansible Project
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+
+DOCUMENTATION = '''
+---
+module: vmware_object_rename
+short_description: Rename a vmware object
+description:
+    - This module can be used to rename a vmware object.
+version_added: 2.6
+author:
+    - Abhijeet Kasurde (@akasurde) <akasurde@redhat.com>
+notes:
+    - Tested on vSphere 6.5
+requirements:
+    - "python >= 2.6"
+    - PyVmomi
+options:
+   name:
+     description:
+     - Name of the object to rename.
+     required: True
+   new_name:
+     description:
+     - New name of the object.
+     required: True
+   object_type:
+     description:
+     - Type of object to rename.
+     required: False
+     default: 'VirtualMachine'
+     choices: ['VirtualMachine', 'Datacenter', 'ClusterComputeResource',
+               'DistributedVirtualSwitch', 'HostSystem', 'Datastore',
+               'Network', 'DistributedVirtualPortgroup']
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = '''
+- name: Rename Datacenter
+  vmware_object_rename:
+    hostname: 192.168.1.209
+    username: administrator@vsphere.local
+    password: vmware
+    validate_certs: False
+    name: datacenter1
+    new_name: datacenter2
+    object_type: 'Datacenter'
+  delegate_to: localhost
+
+- name: Rename Cluster
+  vmware_object_rename:
+    hostname: 192.168.1.209
+    username: administrator@vsphere.local
+    password: vmware
+    validate_certs: False
+    name: cluster1
+    new_name: nw_cluster2
+    object_type: 'ClusterComputeResource'
+  delegate_to: localhost
+'''
+
+RETURN = """ # """
```

Comment:
What does it return?

Reply:
It does not return anything significant apart from changed state.

Reply:
Still it would be better to document that.

Reply:
I agree with @webknjaz 
You should document that if source and destination aren't present the changed state is false. But if the source is absent but the destination is present the module return an error (maybe it shouldn't).